src/code/z_prenmi.c

```diff
@@ -26,26 +25,30 @@ void PreNMI_Draw(PreNMIContext* prenmiCtx) {
     func_8012C470(gfxCtx);
 
     gDPSetFillColor(POLY_OPA_DISP++, (GPACK_RGBA5551(255, 255, 255, 1) << 16) | GPACK_RGBA5551(255, 255, 255, 1));
-    gDPFillRectangle(POLY_OPA_DISP++, 0, prenmiCtx->timer + 100, SCREEN_WIDTH - 1, prenmiCtx->timer + 100);
+    gDPFillRectangle(POLY_OPA_DISP++, 0, this->timer + 100, SCREEN_WIDTH - 1, this->timer + 100);
 
     CLOSE_DISPS(gfxCtx);
 }
 
-void PreNMI_Main(PreNMIContext* prenmiCtx) {
-    PreNMI_Update(prenmiCtx);
-    PreNMI_Draw(prenmiCtx);
+void PreNMI_Main(GameState* thisx) {
+    PreNMIState* this = (PreNMIState*)thisx;
 
-    prenmiCtx->state.unk_A3 = 1;
+    PreNMI_Update(this);
+    PreNMI_Draw(this);
```

Comment:
Should the update and draw functions also take `thisx`?

Reply:
I do not believe so. Gamestate's external facing functions are `Main`, which do take `thisx`, cast, and than call `Update` and `Draw` with the casted variable.